File: backend/utils/playlist.py
import random

from backend.classes.playlist import Playlist


def randomize_playlist(playlist : Playlist, start_index: int = 0) -> list:
    selected_piece = playlist.music_pieces[start_index]
    remaining_pieces = playlist.music_pieces[:start_index] + playlist.music_pieces[start_index + 1:]
    random.shuffle(remaining_pieces)
    queue = [selected_piece] + remaining_pieces
    return queue

File: frontend/src/utils/playlist.js
function shuffleArray(array) {
  let currentIndex = array.length;
  while (currentIndex > 1) {
    const randomIndex = Math.floor(Math.random() * currentIndex);
    currentIndex--;
    const temp = array[currentIndex];
    array[currentIndex] = array[randomIndex];
    array[randomIndex] = temp;
  }
}

export function shufflePlaylist(playlist, startIndex = 0){
  const selectedPiece = playlist[startIndex];
  const remainingPieces = playlist.slice(0, startIndex).concat(playlist.slice(startIndex + 1));
  shuffleArray(remainingPieces);
  return [selectedPiece, ...remainingPieces]
}
